lib/spack/llnl/util/filesystem.py

```diff
@@ -2278,10 +2278,17 @@ def add_rpath(self, *paths):
         """
         self._addl_rpaths = self._addl_rpaths | set(paths)
 
-    def _link(self, path, dest):
+    def _link(self, path, dest_dir):
+        """Perform link step of simulated rpathing, installing
+        simlinks of file in path to the dest_dir
+        location. This method deliberately prevents
+        the case where a path points to a file inside the dest_dir.
+        This is because it is both meaningless from an rpath
+        perspective, and will cause an error when Developer
+        mode is not enabled"""
         file_name = os.path.basename(path)
-        dest_file = os.path.join(dest, file_name)
-        if os.path.exists(dest):
+        dest_file = os.path.join(dest_dir, file_name)
+        if os.path.exists(dest_dir) and not dest_file == path:
```

Comment:
Below here I see `# for py2 compatibility`. We are not supporting Python 2 anymore, so I guess below we can just have:
```
symlink(path, dest_file)
```
? Or is there something I am missing?

Reply:
Thanks for reminding me, we'll still need the try catch block, but now we can directly catch the Python class error as the exception rather than catching all OSErrors and checking the underlying Windows error.